src/util/debug.h

```diff
@@ -5,6 +5,19 @@
 
 #include "errordialoghandler.h"
 
+#ifdef MIXXX_DEBUG_ASSERTIONS_ENABLED
+#define DBG(value)                                                       \
+    (qDebug().nospace() << __FILE__ << ":" << __LINE__ << ": " << #value \
+                        << " = " << value)
```

Comment:
The format would also be up for debate, this can quickly become a bit hard-to-read if the repo is cloned to a path deep in the filesystem. Suggestions are welcome!

Reply:
Indeed, that bothers me with the current debug asserts as well.
A way to remove the root dir down to `src/` would be nice.

Reply:
I would probably keep `src/`, that seems to play a bit better with IDEs when opening the `mixxx` repo there and also handles the case where we hit something in `lib/` code (which should become less likely as we move more dependencies to `vcpkg` though). Using `-fmacro-prefix-map` would relativize all `__FILE__` paths.

Reply:
- #12350